Response/HeaderProvider/CorsVaryHeaderProvider.php

```diff
@@ -63,9 +50,12 @@ public function __construct(
      * ```
      *
      * So, we don't bother attempting to merge the two hears into a comma-separated list.
+     *
+     * //TODO(damienwebdev): consider the ramifications to cache size if we don't
+     * normalize "Vary" this into well organized list.
      */
     public function canApply(): bool
     {
-        return $this->validator->originIsValid() && $this->getValue();
+        return true;
```

Comment:
Where does the origin validation happen? Is this behavior that should be baked intot he module or is it just a special case to handle origins differently?

Reply:
We don't need to validate Origin for the `vary` header at all. We're simply telling all HTTP client's that this header will vary, so use it as a cache key for cache lookups.